bd-session-replay/src/lib.rs

```diff
@@ -133,35 +156,49 @@ impl Recorder {
 
     loop {
       tokio::select! {
-        () = async {
-          if let Some(reporting_interval) = &mut self.reporting_interval {
-            reporting_interval.tick().await;
-          }
-        }, if self.reporting_interval.is_some() && self.is_periodic_reporting_enabled => {
+        () = async { self.reporting_interval.as_mut().unwrap().tick().await; },
+          if self.reporting_interval.is_some() && self.is_periodic_reporting_enabled => {
           log::debug!("session replay recorder capturing wireframe");
           self.target.capture_wireframe();
         },
-        _ = self.reporting_interval_flag.changed() => {
+        _ = self.reporting_interval_rate_flag.changed() => {
```

Comment:
As an aside, as a follow up we might consider removing `read_mark_update` and having `changed` return the new value. This would be a lot less fragile and all the typed implement Copy so there is no perf issue.